make/common/JdkNativeCompilation.gmk

```diff
@@ -199,11 +218,19 @@ define SetupJdkLibraryBody
   endif
 
   ifneq ($$($1_DEFAULT_CFLAGS), false)
-    # Set the default flags first to be able to override
-    $1_CFLAGS := $$(filter-out $$($1_CFLAGS_FILTER_OUT), $$(CFLAGS_JDKLIB)) $$($1_CFLAGS)
+    ifeq ($$($1_TYPE), EXECUTABLE)
+      # Set the default flags first to be able to override
+      $1_CFLAGS := $$(filter-out $$($1_CFLAGS_FILTER_OUT), $$(CFLAGS_JDKEXE)) $$($1_CFLAGS)
 
-    # Set the default flags first to be able to override
-    $1_CXXFLAGS := $$(filter-out $$($1_CXXFLAGS_FILTER_OUT), $$(CXXFLAGS_JDKLIB)) $$($1_CXXFLAGS)
+      # Set the default flags first to be able to override
+      $1_CXXFLAGS := $$(filter-out $$($1_CXXFLAGS_FILTER_OUT), $$(CXXFLAGS_JDKEXE)) $$($1_CXXFLAGS)
+    else
+      # Set the default flags first to be able to override
+      $1_CFLAGS := $$(filter-out $$($1_CFLAGS_FILTER_OUT), $$(CFLAGS_JDKLIB)) $$($1_CFLAGS)
+
+      # Set the default flags first to be able to override
+      $1_CXXFLAGS := $$(filter-out $$($1_CXXFLAGS_FILTER_OUT), $$(CXXFLAGS_JDKLIB)) $$($1_CXXFLAGS)
+    endif
```

Comment:
I think it makes sense to share all that is actually common between the two existing macros, but for these conditional adding default flags, it's just a big if EXECUTABLE do this, otherwise do that. I think in such cases it makes more sense to keep that logic in the respective specialized macros. The only drawback would be that the new `SetupJdkNativeCompilation` won't be usable on its own, but it's not intended to be anyway.

Reply:
Actually, I'm intending on calling SetupJdkNativeCompilation directly in one place, from SetupTestFilesCompilation, which already mixes executable and libraries at it's entry point. So I'd rather keep SetupJdkNativeCompilation working.

Also, I plan to follow this up with a PR that splits up the C[XX]FLAGS_JDK[EXE|LIB] into constituent parts and then just select the few that differs between C and C++, LIB and EXE depending on which is chosen. I should perhaps have been clearer about this.

Reply:
I also see another couple of rounds of refactoring on SetupJdkNativeCompilation and SetupNativeCompilation before they get really into shape. I prefer to do that step by step, so it is possible to follow what changes are being done. If I were to make all changes I'd like and push it in a single PR, it would be very hard to follow how the code has been transformed.